File: CS/DSWizardQueryTables/ConfigureQueryPageViewEx.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.DataAccess.UI.Wizard.Views;
using DevExpress.DataAccess.Wizard.Views;
using DevExpress.Data;
using DevExpress.DataAccess.Wizard.Services;
using DevExpress.DataAccess.Wizard;
using DevExpress.XtraEditors;
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.DataAccess.Native.Sql.QueryBuilder;
using DevExpress.Xpo.DB;

namespace DSWizardCustomizationNoSP
{
    public partial class ConfigureQueryPageViewEx : ConfigureQueryPageView
    {
        internal event EventHandler<SelectedTableChanged> SelectedTableChanged;
        QueryControlView tableQueryControl { get { return panel.Controls[0] as QueryControlView; } }

        public ConfigureQueryPageViewEx(IDisplayNameProvider displayNameProvider, IServiceProvider propertyGridServices, ICustomQueryValidator customQueryValidator, SqlWizardOptions options) :
            base(displayNameProvider, propertyGridServices, customQueryValidator, options)
        {
            InitializeComponent();
            layoutItemQueryType.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never;
            layoutItemQuery.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never;
            layoutItemQueryBuilderButton.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never;
        }

        
        void RaiseSelectedTableChanged(string tableName)
        {
            if (SelectedTableChanged != null)
            {
                SelectedTableChanged(this, new SelectedTableChanged { TableName = tableName });
            }
        }

        internal void InitializeTables(DBSchema dbSchema)
        {
            listBoxDsTables.Items.Clear();
            listBoxDsTables.Items.AddRange(dbSchema.Tables.Select(t => t.Name).ToArray());
        }
        

        private void listBoxDsTables_SelectedIndexChanged(object sender, EventArgs e)
        {
            RaiseSelectedTableChanged((string)listBoxDsTables.SelectedItem);
            //DBTable table = listBoxDsTables.SelectedItem as DBTable;
            //tableQueryControl.SqlString = string.Format("SELECT * FROM {0}", table.Name);
            //base.RaiseChanged();
        }
        
        
    }
    public class SelectedTableChanged : EventArgs
    {
        public string TableName { get; set; }
    }
}

File: VB/DSWizardQueryTables/ConfigureQueryPageViewEx.vb
Imports System
Imports System.Linq
Imports DevExpress.Data
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.UI.Wizard.Views
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Services

Namespace DSWizardCustomizationNoSP
    Partial Public Class ConfigureQueryPageViewEx
        Inherits ConfigureQueryPageView

        Friend Event SelectedTableChanged As EventHandler(Of SelectedTableChanged)
        Private ReadOnly Property tableQueryControl() As QueryControlView
            Get
                Return TryCast(panel.Controls(0), QueryControlView)
            End Get
        End Property

        Public Sub New(ByVal displayNameProvider As IDisplayNameProvider,
                       ByVal propertyGridServices As IServiceProvider,
                       ByVal customQueryValidator As ICustomQueryValidator,
                       ByVal options As SqlWizardOptions)
            MyBase.New(displayNameProvider, propertyGridServices, customQueryValidator, options)
            InitializeComponent()
            layoutItemQueryType.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never
            layoutItemQuery.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never
            layoutItemQueryBuilderButton.Visibility = DevExpress.XtraLayout.Utils.LayoutVisibility.Never
        End Sub


        Private Sub RaiseSelectedTableChanged(ByVal tableName As String)
            RaiseEvent SelectedTableChanged(Me, New SelectedTableChanged With {.TableName = tableName})
        End Sub

        Friend Sub InitializeTables(ByVal dbSchema As DBSchema)
            listBoxDsTables.Items.Clear()
            listBoxDsTables.Items.AddRange(dbSchema.Tables.Select(Function(t) t.Name).ToArray())
        End Sub


        Private Sub listBoxDsTables_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) _
            Handles listBoxDsTables.SelectedIndexChanged
            RaiseSelectedTableChanged(CStr(listBoxDsTables.SelectedItem))
        End Sub


    End Class
    Public Class SelectedTableChanged
        Inherits EventArgs

        Public Property TableName() As String
    End Class
End Namespace

